src/Telegram/Provider.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace SocialiteProviders\Telegram;
+
+use Illuminate\Support\Facades\Validator;
+use InvalidArgumentException;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    const IDENTIFIER = 'TELEGRAM';
+
+    /**
+     * @return array
+     */
+    public static function additionalConfigKeys()
+    {
+        return [
+            'bot',
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        return null;
+    }
+
+    public function getButton()
+    {
+        $botname = $this->config['bot'];
+        $callbackUrl = $this->config['redirect'];
+
+        return '<script async src="https://telegram.org/js/telegram-widget.js" data-telegram-login="'.$botname.'" data-size="large" data-userpic="false" data-auth-url="'.$callbackUrl.'" data-request-access="write"></script>';
```

Comment:
could we use `sprintf` instead of concat here please?

Reply:
Ok, solved in 0735f8d